edit/edit.js

```diff
@@ -0,0 +1,2068 @@
+  const name = document.getElementById('name').value;
+  if (name == '') {
+    return t('styleMissingName');
+  }
+  // validate the regexps
+  if (document.querySelectorAll('.applies-to-list').some(function(list) {
+    return list.childNodes.some(function(li) {
+      if (li.className == template.appliesToEverything.className) {
+        return false;
+      }
+      const valueElement = li.querySelector('[name=applies-value]');
+      const type = li.querySelector('[name=applies-type]').value;
+      const value = valueElement.value;
+      if (type && value) {
+        if (type == 'regexp') {
+          try {
+            new RegExp(value);
+          } catch (ex) {
+            valueElement.focus();
+            return true;
+          }
+        }
+      }
+      return false;
+    });
+  })) {
+    return t('styleBadRegexp');
+  }
+  return null;
+}
+
+function save() {
+  updateLintReport(null, 0);
+
+  // save the contents of the CodeMirror editors back into the textareas
+  for (let i = 0; i < editors.length; i++) {
+    editors[i].save();
+  }
+
+  const error = validate();
+  if (error) {
+    alert(error);
+    return;
+  }
+  const name = document.getElementById('name').value;
+  const enabled = document.getElementById('enabled').checked;
+  saveStyleSafe({
+    id: styleId,
+    name: name,
+    enabled: enabled,
+    reason: 'editSave',
+    sections: getSectionsHashes()
+  })
+    .then(saveComplete);
+}
+
+function getSectionsHashes() {
+  const sections = [];
+  getSections().forEach(function(div) {
+    const meta = getMeta(div);
+    const code = div.CodeMirror.getValue();
+    if (/^\s*$/.test(code) && Object.keys(meta).length == 0) {
+      return;
+    }
+    meta.code = code;
+    sections.push(meta);
+  });
+  return sections;
+}
+
+function getMeta(e) {
+  const meta = {urls: [], urlPrefixes: [], domains: [], regexps: []};
+  e.querySelector('.applies-to-list').childNodes.forEach(function(li) {
+    if (li.className == template.appliesToEverything.className) {
+      return;
+    }
+    const type = li.querySelector('[name=applies-type]').value;
+    const value = li.querySelector('[name=applies-value]').value;
+    if (type && value) {
+      const property = CssToProperty[type];
+      meta[property].push(value);
+    }
+  });
+  return meta;
+}
+
+function saveComplete(style) {
+  styleId = style.id;
+  setCleanGlobal();
+
+  // Go from new style URL to edit style URL
+  if (location.href.indexOf('id=') == -1) {
+    history.replaceState({}, document.title, 'edit.html?id=' + style.id);
+    tE('heading', 'editStyleHeading', null, false);
+  }
+  updateTitle();
+}
+
+function showMozillaFormat() {
+  const popup = showCodeMirrorPopup(t('styleToMozillaFormatTitle'), '', {readOnly: true});
+  popup.codebox.setValue(toMozillaFormat());
+  popup.codebox.execCommand('selectAll');
+}
+
+function toMozillaFormat() {
+  return getSectionsHashes().map(function(section) {
+    let cssMds = [];
+    let i;
```

Comment:
No need to separate the declaration. Also, use `const` because the value isn't changed in its block scope: `for (const i in propertyToCss)`